src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -486,6 +520,86 @@ public function update( $type, $item ) {
 			}
 		}
 
+		if ( 'theme' === $type ) {
+			error_log( '    Theme ' . var_export( $item->theme, true ) . ' has been upgraded.' );
+		}
+
+		if ( 'plugin' === $type ) {
+			error_log( '    Plugin ' . var_export( $item->slug, true ) . ' has been upgraded.' );
+			if ( is_plugin_inactive( $upgrader_item ) ) {
+				error_log( '    ' . var_export( $upgrader_item, true ) . ' is inactive and will not be checked for fatal errors.' );
+			}
+
+			if ( $was_active && ! is_wp_error( $upgrade_result ) ) {
+
+				/*
+				 * The usual time limit is five minutes. However, as a loopback request
+				 * is about to be performed, increase the time limit to account for this.
+				 */
+				if ( function_exists( 'set_time_limit' ) ) {
+					set_time_limit( 10 * MINUTE_IN_SECONDS );
+				}
+
+				// Avoid a race condition when there are 2 sequential plugins that have fatal errors.
+				sleep( 2 );
+
+				if ( $this->has_fatal_error() ) {
+					$upgrade_result = new WP_Error();
+					$temp_backup    = array(
+						array(
+							'dir'  => 'plugins',
+							'slug' => $item->slug,
+							'src'  => WP_PLUGIN_DIR,
+						),
+					);
+
+					$backup_restored = $upgrader->restore_temp_backup( $temp_backup );
+					if ( is_wp_error( $backup_restored ) ) {
+						$upgrade_result->add(
+							'plugin_update_fatal_error_rollback_failed',
+							sprintf(
+								/* translators: %s: The plugin's slug. */
+								__( "The update for '%s' contained a fatal error. The previously installed version could not be restored." ),
+								$item->slug
```

Comment:
Maybe it would be better to also display the full plugin name?
Something like `The update for Plugin Name (plugin-slug) contained a fatal error. The previously installed version could not be restored.`

Reply:
Makes sense! I'll update the PR a little later

Reply:
👍

Reply:
Only concern I see with this is the length of some of the plugin names. As this is only in the error log I think the slug makes more sense.

Reply:
This error actually gets added to the `update_results['plugin']` array rather than being in the error log.

Even still, some plugins do have very long names. For example, not to single it out, it just comes to mind:

> WPSSO Core – The Complete Structured Data Plugin: Schema Markup, Social and SEO Meta Tags, Snippets, Feeds, Sitemaps, Breadcrumbs

@audrasjb What do you think?

Reply:
Yeah that's right. Let's start with the slug. As we're dealing with logs, let's start with this technicval information only 👍